content/graphql/federation.md

```diff
@@ -738,3 +738,132 @@ import { GraphQLModule } from '@nestjs/graphql';
 })
 export class AppModule {}
 ```
+
+### Federation 2
+
+To quote the [Apollo docs](https://www.apollographql.com/docs/federation/federation-2/new-in-federation-2), Federation 2 improves developer experience from the original Apollo Federation (called Federation 1 in this doc), which is backward compatible with most original supergraphs.
+
+> warning **Warning** Mercurius doesn't fully support Federation 2. You can see the list of libraries that support Federation 2 [here](https://www.apollographql.com/docs/federation/supported-subgraphs#javascript--typescript).
+
+In the following sections, we'll upgrade the previous example to Federation 2.
+
+#### Federated example: Users
+
+One change in Federation 2 is that entities has no originating subgraph, so we don't need to extend `Query` anymore. For more detail please refer to [this topic](https://www.apollographql.com/docs/federation/federation-2/new-in-federation-2#entities) in Apollo docs.
+
+#### Schema first
+
+We can simply remove `extend` keyword from the schema.
+
+```graphql
+type User @key(fields: "id") {
+  id: ID!
+  name: String!
+}
+
+type Query {
+  getUser(id: ID!): User
+}
+```
+
+#### Code first
+
+To use Federation 2, we need to specify in `autoSchemaFile` option.
```

Comment:
I don't see anywhere in the Apollo doc mention it as "version 2", but since it follows Apollo releasing stages I think we can assume it's technically a major version 2.